xbmc/GUIInfoManager.cpp

```diff
@@ -7060,6 +7060,15 @@ const infomap fanart_labels[] =  {{ "color1",           FANART_COLOR1 },
 ///     @skinning_v18 **[New Infolabel]** \link Skin_Font `Skin.Font`\endlink
 ///     <p>
 ///   }
+///   \table_row3{   <b>`Skin.Numeric(settingid)`</b>,
```

Comment:
"Numeric" means that something is represented by digits in a positional number system, which is both ints and floats. Are floats currently supported? Is there any value in this?

Reply:
@garbear floats are not supported, the returned value is always an integer (the stored value if it exists, -1 otherwise). I used `Skin.Numeric` because the builtin counterpart is `Skin.SetNumeric`. This is pretty much identical to the existing `Skin.SetString(xx)`/`Skin.String(xx)` and `Skin.SetBool(xxx)`/`Skin.Bool()`. I am trying to implement the close video osd automatically after x configurable settings. The need (and setter) is shown below:

![image](https://user-images.githubusercontent.com/7375276/166141834-9eabf705-2a2e-4f85-9b01-64ba248c81b8.png)

To be able to use this we need to use `Integer` "compare" info expressions. Something like this:

`Integer.IsGreaterOrEqual(Skin.TimerElapsedSecs(autoclosevideoosd),Skin.Numeric(OSDAutoCloseTime))`

The arguments of this infobool are integers, so the value stored in the skin settings needs to be an integer as well. Currently there's also another issue that I still have to tackle: it seems the second info expression parameter is ignored and forced to a static integer value. I mean:

`Integer.IsGreaterOrEqual(Skin.TimerElapsedSecs(autoclosevideoosd),Skin.Numeric(OSDAutoCloseTime))` -> doesn't work

`Integer.IsGreaterOrEqual(Skin.TimerElapsedSecs(autoclosevideoosd),5)` -> works

I am yet to find a way of improving this but I am pretty sure I'll get there :)

Reply:
Would a skin every have any need for a sub-1-second timer? I guess if that ever happens, we can add Ms versions of the functions.

Reply:
hum...it might happen but I think sub ms is too short for achieving anything noticeable (I mean show/hide something) on the interface. Let's see, not hard to add if the use case justifies it.

On the skin timers PR I am using 500ms between evaluations just to avoid pulling them all the time. And because 500ms is usually the interval we use to keep cached values available before pruning (e.g. when you do a player seek the `Player.Seeking` is valid for 500 ms).